packages/block-library/src/editor.scss

```diff
@@ -34,17 +34,222 @@
 @import "./verse/editor.scss";
 @import "./video/editor.scss";
 
-
 /**
  * Editor Normalization Styles
  *
  * These are only output in the editor, but styles here are NOT prefixed .editor-styles-wrapper.
  * This allows us to create normalization styles that are easily overridden by editor styles.
  */
 
+// Reset class selectors that are used on HTML elements.
+// Raise the specificity to keep theme styles from overriding.
+// Specificity = 030.
+:root .editor-styles-wrapper .editor-post-permalink__link,
+:root .editor-styles-wrapper .components-tab-button,
+:root .editor-styles-wrapper .block-editor-url-input > input,
+:root .editor-styles-wrapper .dashicon,
+:root .editor-styles-wrapper .components-button,
+:root .editor-styles-wrapper .components-dropdown-menu__indicator,
+:root .editor-styles-wrapper .block-editor-block-icon {
```

Comment:
Why is the block library stylesheet used for these resets?
How do we decide which component to reset this way?

Reply:
> Why is the block library stylesheet used for these resets?

I actually picked this spot kind of randomly, hoping to get some thoughts and suggestions from you and others.
The specificity is calculated so that the reset can be placed anywhere. 
It could be loaded in a different `editor.scss`, as a separate CSS file, or maybe inlined? Where do you think it belongs?

> How do we decide which component to reset this way?

That's a good question. These were picked out of the browser inspector. 
Manually is the only way I know of to grab the CSS class of every HTML tag used in an editor component.

Something I've been meaning to add here is a utility class so that 3rd party blocks could use the reset in their blocks. Something like `:root .editor-styles-wrapper .editor-protected-tag`.
We could even add the `.editor-protected-tag` class to known core elements rather than listing them individually here.

Any suggestions on naming that ".editor-protected-tag" class?

Reply:
The problem is: 
 
- these components are not the only one that can be used in the editor since any component (block authors are free) can be used in the editor which means it's impossible to know which components to reset. If we were to stay, we limit ourselves to core components, it's not enough as we can add core components without thinking about these resets and it won't be a complete solution.

- the core components (we have hundreds) do not depend on the editor, they don't know that the editor exist, most of them are in a generic package that can be used outside the editor which means we can't add these styles to the components themselves.

What happens if we remove these specific resets and only use the postcss plugin in a generic way?